From 99fc8ab0f6b95889ba0243aea6f1f3d591c21d08 Mon Sep 17 00:00:00 2001
From: Muhammad Husaini Zulkifli <muhammad.husaini.zulkifli@intel.com>
Date: Wed, 28 Apr 2021 11:56:11 +0800
Subject: [PATCH 103/122] igc: Fix fail to set fragment size via ethtool

Fix regression which fail to set fragment size from 256 bytes to 260 bytes
due to data type ranges. Changes the data type range to u32.

Signed-off-by: Muhammad Husaini Zulkifli <muhammad.husaini.zulkifli@intel.com>
---
 drivers/net/ethernet/intel/igc/igc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/intel/igc/igc.h b/drivers/net/ethernet/intel/igc/igc.h
index 824887be2e34..f4f97faf66e0 100644
--- a/drivers/net/ethernet/intel/igc/igc.h
+++ b/drivers/net/ethernet/intel/igc/igc.h
@@ -175,7 +175,7 @@ struct igc_adapter {
 	ktime_t base_time;
 	ktime_t cycle_time;
 	bool frame_preemption_active;
-	u8 add_frag_size;
+	u32 add_frag_size;
 
 	/* OS defined structs */
 	struct pci_dev *pdev;
-- 
2.17.1

